xmtp_mls/src/lib.rs

```diff
@@ -1,3 +1,4 @@
+#![recursion_limit = "256"]
```

Comment:
The recursion limit being high is a result first showed up after adding `tracing::instrument` attribute macros everywhere as a result of benchmarking. These macros are nested within each other depending on the codepath resulting in us reaching the default limit. The easiest is to just remove the tracing::instrument macros. I haven't tried it but we can also specify a [static verbosity level](https://docs.rs/tracing/latest/tracing/level_filters/index.html#compile-time-filters) to skip compiling these macros unless we specify a trace level at compile time. This would probably get rid of the attribute for the `xmtp_mls` crate, but would require the attribute for benchmarks or anything making use of the instrumentation.